### PR TITLE
Fix service host interface to support ts2.5

### PR DESCRIPTION
**Pull request by cevek**
In typescript 2.5 language service host needs to implement fileExists, readFile and readDirectory to be able to correctly resolve the modules (otherwise only ambient modules would be resolved)
Described here https://github.com/Microsoft/TypeScript/blame/master/src/services/types.ts#L166

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -278,6 +278,9 @@ export function register (options: Options = {}): Register {
       },
       getDirectories: getDirectories,
       directoryExists: directoryExists,
+      fileExists: ts.sys.fileExists,
+      readFile: ts.sys.readFile,
+      readDirectory: ts.sys.readDirectory,
```
Comment (blakeembrey) [inline src/index.ts]: We have existing implementations of two of these functions, should we just use those?

Comment (cevek) [inline src/index.ts]: @blakeembrey Fixed

**Comment by johnnyreilly** [comment]
@blakeembrey - do you know if we need to make a similar change to ts-loader here: https://github.com/TypeStrong/ts-loader/blob/master/src/servicesHost.ts#L34 ?

**Comment by blakeembrey** [comment]
@johnnyreilly I would think so, yes. You can just use the `ts.sys` functions though.

**Comment by johnnyreilly** [comment]
Thanks @blakeembrey - I've tested and it looks good.

**Pull request merged**